src/devices/cpu/m6809/6x09dasm.cpp

```diff
@@ -57,42 +57,60 @@ const m6x09_base_disassembler::opcodeinfo *m6x09_base_disassembler::fetch_opcode
 {
 	uint16_t page = 0;
 	const opcodeinfo *op = nullptr;
+
 	while(!op)
 	{
 		// retrieve the opcode
 		uint16_t opcode = page | opcodes.r8(p++);
+		m6x09_instruction_level level = m_level;
 
 		// perform the lookup
 		auto iter = std::find_if(
 			m_opcodes.begin(),
 			m_opcodes.end(),
-			[opcode](const opcodeinfo &info) { return info.opcode() == opcode; });
+			[opcode,level](const opcodeinfo &info) { return (info.opcode() == opcode) && (info.level() <= level); });
```

Comment:
Doing a level comparison like this means you always scan the entire array if there’s no matching opcode.  Have you considered it might be better to use a set with a transparent comparator to avoid the O(n) walks?